### PR TITLE
chore: fix cformat.sh script and use hatch to ensure the same version

### PR DESCRIPTION
**Pull request by juanjux**
## Description

- Move the `cformat` step to hatch so we can `pip install` `clang-format` and ensure the same version.
- Fix an error where it was not parsing `.cpp` files.
- Run it over the codebase.

## Checklist

- [X] Change(s) are motivated and described in the PR description
- [X] Testing strategy is described if automated tests are not included in the PR
- [X] Risks are described (performance impact, potential for breakage, maintainability)
- [X] Change is maintainable (easy to change, telemetry, documentation)
- [X] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [X] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [X] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [X] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by gnufede — APPROVED** [review]

File: scripts/cformat.sh [hunk 1]
```diff
@@ -14,12 +14,23 @@ trap clean EXIT
...
-    THIS_PATH="$(realpath "$0")"
-    THIS_DIR="$(dirname $(dirname "$THIS_PATH"))"
-    for file in $(find "$THIS_DIR" -name '*.[c|cpp|h]' | grep -v '.riot/' | grep -v 'ddtrace/vendor/' | grep -v 'ddtrace/appsec/_iast/_taint_tracking/cmake-build-debug/' | grep -v '^ddtrace/appsec/_iast/_taint_tracking/_vendor/')
-    do
-    clang-format --style="{BasedOnStyle: Mozilla, IndentWidth: 4, ColumnLimit: 120}" -i "$file"
-    done
+  THIS_PATH="$(realpath "$0")"
+  THIS_DIR="$(dirname $(dirname "$THIS_PATH"))"
+  # Find .c, .cpp, and .h files, excluding specified directories
+  find "$THIS_DIR" -type f \( -name '*.c' -o -name '*.cpp' -o -name '*.h' \) \
```
Comment (sanchda) [inline scripts/cformat.sh:20]:
```suggestion
  # Find .c, , .h, .cpp, and .hpp files, excluding specified directories
  find "$THIS_DIR" -type f \( -name '*.c' -o -name '*.h' -o -name '*.cpp' -o -name '*.hpp' \) \
```

**Review by sanchda — APPROVED** [review]
LGTM--thank you!

I added two comments because there are also .hpp files in the repo which should be checked for consistency--I've been manually keeping them up-to-date, so modifying cformat.sh doesn't force any fixups (as far as I can tell).

Adding this isn't urgent though.

File: scripts/cformat.sh [hunk 2]
```diff
@@ -14,12 +14,23 @@ trap clean EXIT
...
+    | grep -v 'ddtrace/vendor/' \
+    | grep -v '_taint_tracking/_vendor/' \
+    | grep -v 'ddtrace/appsec/_iast/_taint_tracking/cmake-build-debug/' \
+    | grep -v '^ddtrace/appsec/_iast/_taint_tracking/_vendor/' \
+    | while IFS= read -r file; do
+  clang-format --style="{BasedOnStyle: Mozilla, IndentWidth: 4, ColumnLimit: 120}" -i $file
+  echo "Formatting $file"
+done
 else
   git ls-files '*.c' '*.cpp' '*.h' | grep -v '^ddtrace/vendor/' | grep -v '^ddtrace/appsec/_iast/_taint_tracking/_vendor/'  | while read filename
```
[the remark this answers is not in the document]
Comment (juanjux) [reply scripts/cformat.sh:35]: Thanks * 2!

File: scripts/cformat.sh (hunk 1 above)
Comment (juanjux) [reply scripts/cformat.sh]: Committed, thanks

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**